storage/storage.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This package provides generic interfaces for pluggable storage engines and associated
+// abstract storage implementation logic
+package storage
+
+import (
+	"context"
+	"time"
+)
+
+// StorageManager interface implemented by associated Manager
+type Manager interface {
+	NewMetricRecord(ctx context.Context, name, query string) (MetricRecord, error)
+	CleanupRecords(ctx context.Context, valid []string) error
+}
+
+// MetricRecord is an interface implemented by StoredMetricRecord.
+type MetricRecord = interface {
```

Comment:
Remove `=`

Reply:
Is this supposed to be different to the MetricRecord interface defined by datastore?

Reply:
No, it is not supposed to be different. I thought interface should be implemented both on the producer and consumer side. I think I'm missing something here - can you elaborate?

Reply:
An interface in Go defines what you *can* or *will* call, not what's available. 

A struct that just has some methods doesn't need to define an interface to say what it does, that's part of the struct itself. Instead you define an interface at the consumer side to say what you're going to call and any struct that has those methods can be used.

Interfaces are also used to abstract away implementation details. For example the io.Writer interface https://godoc.org/io#Writer is defined so that any object that implements that exact Write method can be passed to the various os functions, and the os level doesn't have to care what the object does with that data.

Reply:
Gotcha, this is super-helpful, removed the redundant interface.